lib/ansible/module_utils/facts.py

```diff
@@ -99,6 +99,7 @@ class Facts(object):
                     ('/etc/os-release', 'SuSE'),
                     ('/etc/gentoo-release', 'Gentoo'),
                     ('/etc/os-release', 'Debian'),
+                    ('/etc/debian_version', 'Debian'),
```

Comment:
I have to leave for the day, but I'll work on that tomorrow.
I think I can take a stab at re-factoring it but I'll probably need some
guidance. Could you help me with that?

On Tue, Mar 24, 2015 at 4:16 PM, Toshio Kuratomi <notifications@github.com>
wrote:

> In lib/ansible/module_utils/facts.py
> <https://github.com/ansible/ansible/pull/10527#discussion_r27063079>:
>
> > @@ -99,6 +99,7 @@ class Facts(object):
> >                      ('/etc/os-release', 'SuSE'),
> >                      ('/etc/gentoo-release', 'Gentoo'),
> >                      ('/etc/os-release', 'Debian'),
> > +                    ('/etc/debian_version', 'Debian'),
>
> Ugh -- so this is where I look at the code and think, this needds to be
> refactored to make this new case sane :-).
>
> If you don't want to refactor, here's some pseudo-code for how to do it
> hackily:
>
>  # Try to handle the exceptions now ...if self.facts['distribution_name'] == 'Debian':
>     try:
>         self.get_lsb_facts()
>         self.facts['distribution_release'] = self.facts['lsb']['codename']
>     except:
>         passfor (path, name) in Facts.OSDIST_LIST:
> [...]
>             if self.facts['distribution'] in ('Fedora', ) or (self.facts['distribution'] == 'Debian' and self.facts['distribution_release']  != 'NA':
>
> Like I say pretty hacky. But the code needs a refactor anyway. And we
> aren't putting false values into the data if we do it this way so future
> programmers won't try to figure out what debian-version is doing in
> OSDIST_LIST.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible/pull/10527/files#r27063079>.
>